bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v2/PageImpl.java

```diff
@@ -289,4 +293,19 @@ public String getMainContentSelector() {
         }
         return htmlPageItems;
     }
+    
+    @Override
+	public String getCanonicalURL() {
+		String canonicalURL = null;
+		if (currentPage != null) {
+			String authoredCanonicalURL = pageProperties.get(CANONICAL_URL, String.class);
+			PageManager pageManager = currentPage.getPageManager();
+			if (pageManager != null && pageManager.getPage(authoredCanonicalURL)!=null) {			
```

Comment:
I'm pretty sure `Page#getPageManager()` cannot return null,
Also, the canonical page is resolved twice via `PageManager#getPage()`. 
Once to assert the page exists and perform the null check, a second time to actually use it to craft the URL. 

Although we can be pretty certain that the second call won't return null, there is no guarantee.

An alternative would be something like 
```JAVA
    @Override
	public String getCanonicalURL() {
        if (currentPage != null) {
            String authoredCanonicalURL = pageProperties.get(CANONICAL_URL, String.class)
            PageManager pageManager = currentPage.getPageManager();
            return Optional.ofNullable(authoredCanonicalURL)
                .map(pageManager::getPage)
                .map(p -> Utils.getURL(request, pageManager, p))
                .orElse(authoredCanonicalURL);
        }
        return null
    }
```

Reply:
Made changes as suggested.

Reply:
@vladbailescu Did you get a chance to review the changes? These changes taken care upon your review comments?